hazelcast/src/hazelcast/util/Thread.cpp

```diff
@@ -61,8 +61,8 @@ namespace hazelcast {
             if (!isJoined) {
                 cancel();
                 join();
+                CloseHandle(thread);
```

Comment:
Should not we close the handle if thread is already joined when destructor called ?

Reply:
If the thread is joined already this means that the above code inside if statement executed, hence the handle is closed. Closing the handle twice it may cause as problem as indicated here: https://msdn.microsoft.com/en-us/library/windows/desktop/ms724211(v=vs.85).aspx
"the function will throw an exception if it receives either a handle value that is not valid or a pseudo-handle value. This can happen if you close a handle twice"